Suppress unimplemented MSR related warnings

brendabrandy opened the pull request:
While using bhyve on intel machines, if certain MSRs are unimplemented, the bhyve console is spammed with these messages: 

```
rdmsr to register 0x684 on vcpu 1
rdmsr to register 0x6c4 on vcpu 1
rdmsr to register 0x683 on vcpu 1
rdmsr to register 0x6c3 on vcpu 1
rdmsr to register 0x682 on vcpu 1
rdmsr to register 0x6c2 on vcpu 1
rdmsr to register 0x681 on vcpu 1
rdmsr to register 0x6c1 on vcpu 1
rdmsr to register 0x680 on vcpu 1
rdmsr to register 0x6c0 on vcpu 1
Password:
Login incorrect
login:
login:
login:
login: rdmsr to register 0x1c9 on vcpu 0
rdmsr to register 0x1dd on vcpu 0
rdmsr to register 0x1de on vcpu 0
rdmsr to register 0x68f on vcpu 0
rdmsr to register 0x6cf on vcpu 0
rdmsr to register 0x68e on vcpu 0
rdmsr to register 0x6ce on vcpu 0
rdmsr to register 0x68d on vcpu 0
rdmsr to register 0x6cd on vcpu 0
rdmsr to register 0x68c on vcpu 0
rdmsr to register 0x6cc on vcpu 0
rdmsr to register 0x68b on vcpu 0
rdmsr to register 0x6cb on vcpu 0
rdmsr to register 0x68a on vcpu 0
rdmsr to register 0x6ca on vcpu 0
rdmsr to register 0x689 on vcpu 0
rdmsr to register 0x6c9 on vcpu 0
rdmsr to register 0x688 on vcpu 0
rdmsr to register 0x6c8 on vcpu 0
rdmsr to register 0x687 on vcpu 0
rdmsr to register 0x6c7 on vcpu 0
rdmsr to register 0x686 on vcpu 0
```
This does not interfere with the normal operation of bhyve (since I can run the image), and there are already options to ignore unimplmented MSRs and continue (`-w`)   But the warning messages makes it difficult to use the console. A new option `-q` is created to suppress the `rdmsr` and `wrmsr` messages.

bsdjhb commented on usr.sbin/bhyve/amd64/bhyverun_machdep.c:
```diff
@@ -96,6 +97,7 @@ bhyve_usage(int code)
 #ifdef BHYVE_SNAPSHOT
 	    "       -r: path to checkpoint file\n"
 #endif
+	    "       -q: disable verbose MSR print out\n"
```
I would probably not add a new top-level option as there are already several, and -q is pretty generic.  I think having it just settable via '-o' is fine.

brendabrandy replied:
thanks for the feedback, i made it settable via o instead now, and added the corresponding entries in the man pages

bsdjhb commented on usr.sbin/bhyve/amd64/bhyverun_machdep.c:
```diff
@@ -63,6 +63,7 @@ bhyve_init_config(void)
 	set_config_bool("acpi_tables_in_memory", true);
 	set_config_value("memory.size", "256M");
 	set_config_bool("x86.strictmsr", true);
+	set_config_bool("x86.verbosemsr", true);
```
I think we can actually default it to false, only developers probably care and for users it is just annoying spam.

brendabrandy replied:
I chanaged this to false now

bsdjhb commented on usr.sbin/bhyve/bhyve.8:
```diff
@@ -295,6 +295,13 @@ To map a 4 vCPU guest to host CPUs 12-15:
 .Bd -literal
 -p 0:12 -p 1:13 -p 2:14 -p 3:15
 .Ed
+.It Fl q
```
If we don't do -q, we should still document this in bhyve_config.5 which describes all the config nodes.

brendabrandy replied:
updated this as well!

ocochard commented:
Tested with and without the `-o x86.verbosemsr=true` and it is working great: Thanks

bsdjhb commented:
The change looks good to me.  Can you rebase it and collapse it down to a single commit with an updated log message?

brendabrandy commented:
> The change looks good to me. Can you rebase it and collapse it down to a single commit with an updated log message?

Thanks for the review! I rebased and squashed those commits, but it looks like the jenkins build failed with "Monthly compute limit exceeded error". Can we still merge or should I kick off the build again somehow

brendabrandy commented:
Thanks for the approvals! I do need someone to help me merge this PR because this is my first PR and I don't have permission to merge :D

bsdjhb reviewed (approved):
A few notes on the commit log when this is merged:

- Use a "bhyve: " prefix for the title
- s/emits/can emit/, s/unimplemented MSR/unimplemented MSRs/
- Maybe s/a lot of/many/
